gazebo_ros/scripts/spawn_model.py

```diff
@@ -0,0 +1,312 @@
+            self.get_logger().error('Invalid XML: {}'.format(e))
+            return 1
+
+        # Replace package:// with model:// for mesh tags if flag is set
+        if self.args.package_to_model:
+            for element in xml_parsed.iterfind('.//mesh'):
+                filename_tag = element.get('filename')
+                if filename_tag is None:
+                    continue
+                url = urlsplit(filename_tag)
+                if url.scheme == 'package':
+                    url = SplitResult('model', *url[1:])
+                    element.set('filename', url.geturl())
+
+        # Encode xml object back into string for service call
+        model_xml = ElementTree.tostring(xml_parsed)
+
+        # Form requested Pose from arguments
+        initial_pose = Pose()
+        initial_pose.position.x = float(self.args.x)
+        initial_pose.position.y = float(self.args.y)
+        initial_pose.position.z = float(self.args.z)
+
+        q = quaternion_from_euler(self.args.R, self.args.P, self.args.Y)
+        initial_pose.orientation.w = q[0]
+        initial_pose.orientation.x = q[1]
+        initial_pose.orientation.y = q[2]
+        initial_pose.orientation.z = q[3]
+
+        # Spawn model using urdf or sdf service based on arguments
+        success = self._spawn_entity(model_xml, initial_pose)
+        if not success:
+            self.get_logger().error('Spawn service failed. Exiting.')
+            return 1
+
+        # TODO(shivesh): Wait for /set_model_configuration
+        # (https://github.com/ros-simulation/gazebo_ros_pkgs/issues/779)
+        # Apply joint positions if any specified
+        # if len(self.args.joints) != 0:
+        #     joint_names = [joint[0] for joint in self.args.joints]
+        #     joint_positions = [joint[1] for joint in self.args.joints]
+        #     success = _set_model_configuration(joint_names, joint_positions)
+        #     if not success:
+        #         self.get_logger().error('SetModelConfiguration service failed. Exiting.')
+        #         return 1
+
+        # Unpause physics if user requested
+        if self.args.unpause:
+            client = self.create_client(Empty, '/unpause_physics')
+            client.wait_for_service(timeout_sec=5)
+            self.get_logger().info('Calling service /unpause_physics')
+            client.call_async(Empty.Request())
+
+        # TODO(shivesh): Wait for https://github.com/ros2/rclpy/issues/244
+        # If bond enabled, setup shutdown callback and wait for shutdown
+        # if self.args.bond:
+        #     rospy.on_shutdown(self._delete_entity)
+        #     rospy.loginfo('Waiting for shutdown to delete model {}'.format(self.args.model))
+        #     rospy.spin()
+
+        return 0
+
+    def _spawn_entity(self, model_xml, initial_pose):
+        self.get_logger().info('Waiting for service %s/spawn_entity' % self.args.gazebo_namespace)
+        client = self.create_client(SpawnEntity, 'spawn_entity')
```

Comment:
The messages include `gazebo_namespace`, but the client doesn't actually use it. I think we should use it, and change the default value to empty, because that's the default namespace offered by the plugins on the other end.